File: 1019-SquaresOfASortedArray/1019-SquaresOfASortedArray.py
# Last updated: 9/14/2025, 9:11:32 AM
import math
#using two pointers
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
            l,r=0,len(nums)-1
            res=[0 for i in nums]
            for i in range(len(nums)-1,-1,-1):
                if abs(nums[l])>abs(nums[r]):
                    res[i]=nums[l]*nums[l]
                    l+=1
                else:
                    res[i]=nums[r]*nums[r]
                    r-=1
            return res

File: 1019-SquaresOfASortedArray/1019-SquaresOfASortedArray.java
// Last updated: 9/14/2025, 9:11:37 AM
//solve this using two pointer
class Solution {
    public int[] sortedSquares(int[] nums) {
        int l=0,r=nums.length-1,square;
        int []res=new int[nums.length];
        for(int i=nums.length-1;i>=0;i--){
             if(Math.abs(nums[l])>Math.abs(nums[r])){
                   square=nums[l];
                   l++;
             }else{
                 square=nums[r];
                 r--;
             }
             res[i]=square*square;
        }
        return res;
    }
}